idstore.go

```diff
@@ -0,0 +1,86 @@
+package blockstore
+
+import (
+	"context"
+
+	blocks "github.com/ipfs/go-block-format"
+	cid "github.com/ipfs/go-cid"
+	mh "github.com/multiformats/go-multihash"
+)
+
+// idstore wraps a BlockStore to add support for identity hashes
+type idstore struct {
+	bs Blockstore
+}
+
+func IdStore(bs Blockstore) Blockstore {
+	return &idstore{bs}
+}
+
+func extractContents(k *cid.Cid) (bool, []byte) {
+	dmh, err := mh.Decode(k.Hash())
+	if err != nil || dmh.Code != mh.ID {
+		return false, nil
+	}
+	return true, dmh.Digest
+}
+
+func (b *idstore) DeleteBlock(k *cid.Cid) error {
+	isId, _ := extractContents(k)
+	if isId {
+		return nil
+	}
+	err := b.bs.DeleteBlock(k)
+	if isId && err == ErrNotFound {
```

Comment:
Looks redundant, `isId` should always be false.

Reply:
Yeah, will fix.